server/data/data-jmap-cassandra/src/main/java/org/apache/james/jmap/cassandra/vacation/CassandraVacationDAO.java

```diff
@@ -36,51 +37,53 @@
 import org.apache.james.backends.cassandra.utils.ZonedDateTimeRepresentation;
 import org.apache.james.jmap.api.vacation.AccountId;
 import org.apache.james.jmap.api.vacation.Vacation;
+import org.apache.james.jmap.api.vacation.VacationPatch;
 import org.apache.james.jmap.cassandra.vacation.tables.CassandraVacationTable;
+import org.apache.james.util.FunctionGenerator;
+import org.apache.james.util.MissableOptional;
 
 import com.datastax.driver.core.PreparedStatement;
 import com.datastax.driver.core.Row;
 import com.datastax.driver.core.Session;
 import com.datastax.driver.core.UDTValue;
 import com.datastax.driver.core.UserType;
+import com.datastax.driver.core.querybuilder.Insert;
+import com.google.common.collect.ImmutableList;
 
 public class CassandraVacationDAO {
 
     private final CassandraAsyncExecutor cassandraAsyncExecutor;
-    private final PreparedStatement insertStatement;
     private final PreparedStatement readStatement;
     private final UserType zonedDateTimeUserType;
+    private final FunctionGenerator<VacationPatch, Insert> insertGeneratorPipeline;
 
     @Inject
     public CassandraVacationDAO(Session session, CassandraTypesProvider cassandraTypesProvider) {
         this.zonedDateTimeUserType = cassandraTypesProvider.getDefinedUserType(CassandraZonedDateTimeModule.ZONED_DATE_TIME);
         this.cassandraAsyncExecutor = new CassandraAsyncExecutor(session);
 
-        this.insertStatement = session.prepare(insertInto(CassandraVacationTable.TABLE_NAME)
-            .value(CassandraVacationTable.ACCOUNT_ID, bindMarker(CassandraVacationTable.ACCOUNT_ID))
-            .value(CassandraVacationTable.IS_ENABLED, bindMarker(CassandraVacationTable.IS_ENABLED))
-            .value(CassandraVacationTable.FROM_DATE, bindMarker(CassandraVacationTable.FROM_DATE))
-            .value(CassandraVacationTable.TO_DATE, bindMarker(CassandraVacationTable.TO_DATE))
-            .value(CassandraVacationTable.TEXT, bindMarker(CassandraVacationTable.TEXT))
-            .value(CassandraVacationTable.SUBJECT, bindMarker(CassandraVacationTable.SUBJECT))
-            .value(CassandraVacationTable.HTML, bindMarker(CassandraVacationTable.HTML)));
-
         this.readStatement = session.prepare(select()
             .from(CassandraVacationTable.TABLE_NAME)
             .where(eq(CassandraVacationTable.ACCOUNT_ID,
                 bindMarker(CassandraVacationTable.ACCOUNT_ID))));
+
+        insertGeneratorPipeline = ImmutableList.<FunctionGenerator<VacationPatch, Insert>>of(
```

Comment:
you could start by defining mapping between fields and getters with a map : 
```
ImmutableMap.of(CassandraVacationTable.SUBJECT, VacationPatch::getSubject)
```
And then you could create the query in the collector